x-pack/metricbeat/module/azure/compute_vm_scaleset/compute_vm_scaleset.go

```diff
@@ -47,6 +47,15 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "error unpack raw module config using UnpackConfig")
 	}
+
+	// validate config resource options entered, no resource queries allowed
+	for _, resource := range config.Resources {
+		if resource.Query != "" {
+			return nil, errors.New("error initializing the monitor client: module azure - compute_vm_scaleset metricset. No queries allowed, please select one of the allowed options")
+		}
+	}
```

Comment:
Could this code be somehow reused? It is used by both metricsets.

Reply:
I've created an azure metricset and pushed all common code for all 3 metricsets